src/QMCWaveFunctions/Fermion/DiracDeterminant.cpp

```diff
@@ -131,9 +134,16 @@ DiracDeterminant::ratioGrad(ParticleSet& P, int iat, GradType& grad_iat)
   const int WorkingIndex = iat-FirstIndex;
   UpdateMode=ORB_PBYP_PARTIAL;
   GradType rv;
-  curRatio = updateEng.ratioGrad(psiM, WorkingIndex, psiV, dpsiV, rv);
-  grad_iat += ((RealType)1.0/curRatio) * rv;
-  RatioTimer.stop();
+
+  // check invRow_id against WorkingIndex to ensure getInvRow() called before ratioGrad()
+  // This is not a safety check. Some code paths do call ratioGrad without calling evalGrad first.
```

Comment:
Commentary: Unfortunate that this is needed, but how would anyone ever know to call evalGrad before calling ratioGrad?

Reply:
Good point. I have update the comment which is more clear as an optimization.